File: DotNetPractices/ProblemSolving/Codility/FrogRiverOne.cs

class FrogRiverOne
{
    public int Solution(int X, int[] A)
    {
        int c = 0, v;
        var b = new bool[A.Length];

        for (int i = 0; i < A.Length; i++)
        {
            v = A[i];
            if (v <= X && !b[v - 1])
            {
                b[v - 1] = true;
                c++;
                if (c == X) return i;
            }
        }

        return -1;
    }
}

File: DotNetPractices/ProblemSolving/c/frogRiverOne.c
#include <string.h>

int frogRiverOne(int X, int A[], int N) {
    int c=0, v;
    char b[X];
    memset(b, 0, X);

    for(int i=0;  i<N; i++) {
        v = A[i];
        if(v <= X && b[v-1] != '1' ) {
            b[v-1] = '1';
            c++;
            if(c==X) return i;
        }
    }

    return -1;
}
